pkg/v1/remote/descriptor.go

```diff
@@ -237,6 +239,49 @@ func (f *fetcher) url(resource, identifier string) url.URL {
 	}
 }
 
+// https://github.com/opencontainers/distribution-spec/blob/main/spec.md#referrers-tag-schema
+func fallbackTag(d name.Digest) name.Tag {
+	return d.Context().Tag(fmt.Sprintf("%s-%s", d.Digest().Algorithm, d.Digest().Hex))
+}
+
+func (f *fetcher) fetchReferrers(ctx context.Context, filter map[string]string, d name.Digest) (*v1.IndexManifest, *ReferrersNextPage, error) {
+	// The registry doesn't support the Referrers API endpoint, so we'll use the fallback tag scheme.
+	b, _, err := f.fetchManifest(fallbackTag(d), []types.MediaType{types.OCIImageIndex})
+	if err != nil {
+		return nil, nil, err
+	}
+	var terr *transport.Error
+	if ok := errors.As(err, &terr); ok && terr.StatusCode == http.StatusNotFound {
+		// Not found just means there are no attachments yet. Start with an empty manifest.
+		return nil, nil, nil
+	}
+	// TODO: What am I supposed to do with a 400 here?
+	var im v1.IndexManifest
+	if err := json.Unmarshal(b, &im); err != nil {
+		return nil, nil, err
+	}
+
+	// If filter applied, filter out by artifactType and add annotation
+	// See https://github.com/opencontainers/distribution-spec/blob/main/spec.md#listing-referrers
+	if filter != nil {
```

Comment:
was not sure if wanting to actually filter the contents of the fallback tag for the client, but would like to at least start using something like `WithFilter()`. Can also be dropped for later

Reply:
I like the filtering, let's keep it.

It might make sense as a separate method we can unit test separately, and maybe eventually export separately in a future change.

We also might be able to lean on https://pkg.go.dev/github.com/google/go-containerregistry@v0.13.0/pkg/v1/match